torch/_prims/__init__.py

```diff
@@ -398,7 +398,8 @@ def _elementwise_meta(
             else:
                 dtype = dtype
 
-        return TensorMeta(device=device, shape=shape, strides=strides, dtype=dtype)
+        assert shape is not None
+        return torch.empty_permuted(shape, l2p_perm, device=device, dtype=dtype)  # type: ignore[return-value]
```

Comment:
can you explain why this changed? 

The headerdoc for this function also mentions strides, do we need to update it?

Reply:
Previously we call `empty_strided`, which triggers a bunch of guards from looking at the strides and working out if they're contiguous or not. This `empty_permuted` bypasses that as it's guaranteed to be contiguous. I don't think this needs to change.